File: app/src/main/java/nz/co/codebros/quakesnz/presenter/BasePresenter.java
package nz.co.codebros.quakesnz.presenter;

import android.os.Bundle;

import java.util.ArrayList;

import io.reactivex.disposables.Disposable;

/**
 * Created by leandro on 19/06/17.
 */

public abstract class BasePresenter {
    private final ArrayList<Disposable> disposables = new ArrayList<>();

    protected void addDisposable(Disposable disposable) {
        disposables.add(disposable);
    }

    protected void disposeAll(){
        while (!disposables.isEmpty()) {
            disposables.remove(0).dispose();
        }
    }

    public abstract void onViewRestored(Bundle bundle);

    public abstract void onViewCreated();

    public abstract void onSaveInstanceState(Bundle bundle);
}

File: app/src/main/java/nz/co/codebros/quakesnz/presenter/BasePresenter.kt
package nz.co.codebros.quakesnz.presenter

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

/**
 * Created by leandro on 19/06/17.
 */

abstract class BasePresenter<out V : View>(
        val view: V
) : Presenter {
    private val disposables = CompositeDisposable()

    protected fun addDisposable(disposable: Disposable) {
        disposables.add(disposable)
    }

    protected fun disposeAll() {
        disposables.dispose()
    }
}
